### PR TITLE
W-13593951: Create polling for dynamically change tracing conf according to file

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

File: tracing/src/test/java/org/mule/test/components/tracing/ExportConfigurationChangeTestCase.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+  @ClassRule
+  public static final TestServerRule originalServer = new TestServerRule();
+
+  @ClassRule
+  public static final TestServerRule afterConfigurationChangeServer = new TestServerRule();
+
+  @Override
+  protected void doSetUpBeforeMuleContextCreation() throws Exception {
+    file = createTempFile("tracing", "test");
+    configFile = Paths.get(getClass().getResource("/tracing/config/exporter.conf").getPath());
```
Comment (sofiamorseletto) [inline tracing/src/test/java/org/mule/test/components/tracing/ExportConfigurationChangeTestCase.java:97]: Will this work on Windows too?

File: tracing/src/test/java/org/mule/test/components/tracing/ExportConfigurationChangeTestCase.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+    Collection<CapturedExportedSpan> exportedSpans = originalServer.getCapturedExportedSpans();
+
+    Map<String, String> setPayloadAttributeMap = createAttributeMap(SET_PAYLOAD_LOCATION, TEST_ARTIFACT_ID);
+
+    assertExpectedSpanTree(attributesToAssertExistence, exportedSpans, setPayloadAttributeMap);
+
+    // We update the file by recopying it.
+    setProperty(MULE_OPEN_TELEMETRY_EXPORTER_ENDPOINT,
+                "http://localhost:" + afterConfigurationChangeServer.httpPort());
```
Comment (sofiamorseletto) [inline tracing/src/test/java/org/mule/test/components/tracing/ExportConfigurationChangeTestCase.java:130]: Does the httpPort change from the original one? Isn't it the same?

Comment (fsgonz) [reply tracing/src/test/java/org/mule/test/components/tracing/ExportConfigurationChangeTestCase.java:134]: Changed to make sure that this works in all testing envs.

File: tracing/src/test/java/org/mule/test/components/tracing/ExportConfigurationChangeTestCase.java (hunk 1 above)
Comment (fsgonz) [reply tracing/src/test/java/org/mule/test/components/tracing/ExportConfigurationChangeTestCase.java]: Changed to make sure that this works in all testing envs.

File: tracing/src/test/java/org/mule/test/components/tracing/ExportConfigurationChangeTestCase.java (hunk 2 above)
Comment (fsgonz) [reply tracing/src/test/java/org/mule/test/components/tracing/ExportConfigurationChangeTestCase.java:134]: Yes, the port to which export also changes.

Comment (fsgonz) [reply tracing/src/test/java/org/mule/test/components/tracing/ExportConfigurationChangeTestCase.java:134]: Basically there are two servers that are created for the export of the telemetry. After the change in the file it begins to export the other server (collector).

**Review by sofiamorseletto — APPROVED** [review]

**Comment by fsgonz** [comment]
--validate

**Pull request merged**